app/parse.py

```diff
@@ -2,11 +1,132 @@
+import csv
+import time
+
+import requests
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from dataclasses import dataclass, fields
+from bs4 import BeautifulSoup
+
+BASE_URL = "https://mate.academy"
+
+
+class Driver:
+    Instance = None
+
+    @staticmethod
+    def initialize() -> webdriver.Chrome:
+        if Driver.Instance is None:
+            Driver.Instance = webdriver.Chrome()
+        return Driver.Instance
 
 
 @dataclass
 class Course:
     name: str
     short_description: str
     duration: str
+    num_topics: str
+    num_modules: str
+
+
+def parse_single_course(soup: BeautifulSoup) -> Course:
+    name = soup.select_one(".ProfessionCard_title__Zq5ZY").text
+    short_description = soup.select_one(
+        ".ProfessionCard_cardWrapper__JQBNJ >"
+        " p.typography_landingTextMain__Rc8BD"
+    ).text
+    duration = soup.select(
+        ".ProfessionCard_professionTags__kCD1H > "
+        "p.ProfessionCardTags_regularTag__yTc6K"
+    )[-1].text
+
+    details = soup.select_one(".typography_landingH3__vTjok")["href"]
+    course_details_url = BASE_URL + details
+
+    num_topics = parse_num_topics(course_details_url)
+    num_modules = parse_num_modules(course_details_url)
+
+    return Course(
+        name=name,
+        short_description=short_description,
+        duration=duration,
+        num_topics=num_topics,
+        num_modules=num_modules
+    )
+
+
+def parse_num_topics(page_url: str) -> str:
+    page = requests.get(page_url).content
+    soup = BeautifulSoup(page, "html.parser")
+
+    num_topics = soup.select_one(
+        ".CourseProgram_purple__STsuM > .FactBlock_factNumber__d_8nn"
+    ).text
+
+    return num_topics
+
+
+def parse_num_modules(page_url: str) -> str:
+    driver = Driver.initialize()
```

Comment:
It seems like you are initializing a new WebDriver for every call to `parse_num_modules`. This can be quite inefficient as it involves a lot of overhead. Consider initializing the WebDriver once and reusing it for multiple calls.